templates-java/src/main/resources/templates/config/template-producer-message.txt

```diff
@@ -0,0 +1,39 @@
+package ${package};
+
+import com.amazonaws.services.sqs.AmazonSQSAsync;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cloud.aws.messaging.core.QueueMessagingTemplate;
+import org.springframework.cloud.aws.messaging.core.SqsMessageHeaders;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+@Component
+public class ProducerMessage {
+
+    @Value("${cloud.aws.sqs.queue-name}")
+    private String queueName;
+
+    private QueueMessagingTemplate queueMessagingTemplate;
+
+    @Autowired
+    public ProducerMessage(AmazonSQSAsync amazonSqs) {
```

Comment:
Consider using constructor-based injection for all dependencies, including `queueName`, to make the class more testable and to avoid using `@Value` for simple string values.